pkg/reconciler/ingress/ingress.go

```diff
@@ -234,7 +235,8 @@ func (c *Controller) desiredLeaves(ctx context.Context, root *networkingv1.Ingre
 	}
 
 	// TODO(jmprusi): Calculate the max string length.
-	generateName := strings.Split(rootKey, "#$#")[1] + "-"
+	_, rootIngressName := clusters.SplitClusterAwareKey(rootKey)
```

Comment:
I think you need to call cache.SplitMetaNamespaceKey first, to get namespace, clusterAwareName, err. Then you can call SplitClusterAwareKey(clusterAwareName) to get cluster, name.

Reply:
Wait, you already have `root` that you can use, can't you? `root.Name`?

Reply:
True! fixed